app/coffee/LockManager.coffee

```diff
@@ -37,8 +39,11 @@ module.exports = LockManager =
 			return callback(err) if err?
 			if gotLock == "OK"
 				metrics.inc "doc-not-blocking"
-				profile.log("got lock").end()
-				callback err, true, lockValue
+				timeTaken = profile.log("got lock").end()
+				if timeTaken > MAX_REDIS_REQUEST_LENGTH
+					callback err, false # took too long to get the lock, bail out
```

Comment:
We have set the lock, even though it took too long, so we shouldn't return false here. We need to expire it if we're going to return false, or it will be blocked for `LOCK_TTL`. Should we instead free the lock and return an error - that feels more semantic. Also, I know it's not your code, but having the `err` in the callback trips me up, since at this point there is no error (it's null, as per the test on the first line of the callback).